R/compare.R

```diff
@@ -101,12 +146,54 @@ compare <- function(x, y, ...,
     ignore_attr = ignore_attr,
     ignore_encoding = ignore_encoding,
     ignore_formula_env = ignore_formula_env,
-    ignore_function_env = ignore_function_env
+    ignore_function_env = ignore_function_env,
+    ignore_name_order = ignore_name_order,
+    ignore_NULLs = ignore_NULLs,
+    ignore_private = ignore_private
   )
+
+  opts$user_specified <- c(intersect(names(match.call()),
+                                   names(opts)),
+                           "user_specified")
+
   out <- compare_structure(x, y, paths = c(x_arg, y_arg), opts = opts)
   new_compare(out, max_diffs)
 }
 
+sort_named_parts <- function(x) {
+  nx <- names(x)
```

Comment:
You can use `names2()` to simplify this — I think `x[order(names2(x))]` will be equivalent (since `order()` is stable, and it's probably fine to sort unnamed components together).

Reply:
I switched to `rlang::names2`:  did you mean to import it in the namespace?

I didn't simplify the way you suggested.  My thinking was that if someone has some unnamed components, then they might have been referring to them by index rather than name, so if a difference shows up, it would be confusing to report it with a different index.